File: Week 2/G-Social Distance.cpp
#include<iostream>
using namespace std;
void solve(){
	long long n,m;
	cin>>n>>m;
	long long a[n];
	
	for(long long i=0;i<n;i++){
		cin>>a[i];
	}
	sort(a,a+n);
	m-=1;
	for(long long i=1;i<n;i++){
		m-=(a[i]+1);
	}
	if(m>=a[n-1]){
	   cout<<"YES"<<endl;
	}else{
	   cout<<"NO"<<endl;
	}
 
}
int main(){
	int t;
	cin>>t;
	while(t--){
		solve();
	}
	
	return 0;
}

File: Week 2/G-Social Distance.py
t = int(input())
while t:
    t-=1
    n,m = map(int, input().split())
    a = list(map(int,input().split()))
    a.sort()
    if n==2:
        if 2*(a[1]+1)<=m:
            print("YES")
        else:
            print("NO")
        continue
    x = a[n-1]+2
    y = m-a[n-1]
    for i in range(n-2, 1, -1):
        if i%2:
            x+=(a[i]+1)
        else:
            y = y-(a[i]+1)
    if y-x > a[1]:
        print("YES")
    else:
        print("NO")